platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationContext.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+/**
+ * The migration context is defined as a base interface for all migration contexts.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface MigrationContext {
```

Comment:
My 2 cents on this. I do agree that there is overlap between this and the `Configurator`. However, I think that until we have a better grasp of how we want to evolve the installation, configuration, migration and upgrade capabilities, it will be difficult to rework this _and_ the `Configurator` and still get all of it ready for the upcoming release. So for now, I'm ok with the duplication and the fact that we will most likely have to rework some of this code and the `Configurator` in the future to get the where we need to be.

@paouelle and @coyotesqrl, thoughts?